plugins/mac-capture/mac-window-capture.m

```diff
@@ -102,6 +104,13 @@ static inline void capture_frame(struct window_capture *wc)
 
 	da_init(wc->buffer);
 
+	blog(LOG_INFO, "[window-capture] - Init Display Capture for permissions dialog");
+	
+	void *dc = display_capture_create(settings, source);
```

Comment:
Makes call to https://developer.apple.com/documentation/coregraphics/1454968-cgdisplaystreamcreatewithdispatc?language=objc whichs also pops up the permissions screen recording dialog

Reply:
does we need a "dc" later or can release it here?

Reply:
no, only for null check.

Reply:
then display_capture_destroy to not leak memory. 

is there a way to check for permission without creating and initializing display capture?

Reply:
not that I know of.

Reply:
yes. looks like it is a way of macos permissions. you need to create display object to trigger permission request. 
maybe not reuse display_capture_create but create new function what only call CGDisplayStreamCreate with minimum parameters and release.

Reply:
@summeroff Reordered the code a bit more to have less exposure to that display_capture_create function, retested it and it seems to work. Of course if also Trevor can retest that would be best even.

Reply:
@summeroff could you update this conversation: should it be marked as resolved or do you still have some comments?